### PR TITLE
docs: add detailed api reference for all sources

### PR DESCRIPTION
**Pull request by nikhilgarg28**

### CODE REVIEW

File: fennel/_vendor/README.md
[the remark this answers is not in the document]
Comment (nikhilgarg28) [reply fennel/_vendor/README.md:1]: Done. Thanks!

File: fennel/datasets/datasets.py [hunk 1]
```diff
@@ -1339,7 +1339,7 @@ def signature(self):
     def with_source(
         self,
         conn: DataConnector,
-        every: Optional[Duration] = None,
+        every: Duration = "1h",
```
Comment (nonibansal) [inline fennel/datasets/datasets.py:1342]: Any reason behind 1h, I feel like 1D should be a good default.

File: fennel/lib/to_proto/to_proto.py [hunk 2]
```diff
@@ -691,8 +689,7 @@ def _kafka_to_ext_db_proto(
             sasl_plain_username=sasl_plain_username,
             sasl_plain_password=sasl_plain_password,
             sasl_jaas_config="",
-            # TODO(xiao): remove once server side is updated
-            enable_ssl_certificate_verification=BoolValue(value=verify_cert),
+            enable_ssl_certificate_verification=None,
```
Comment (nonibansal) [inline fennel/lib/to_proto/to_proto.py:692]: Is this works? What if we pass false as default?

File: fennel/datasets/datasets.py (hunk 1 above)
Comment (nonibansal) [reply fennel/datasets/datasets.py:1342]: Oh alright.

**Pull request merged**